src/aiida/cmdline/commands/cmd_profile.py

```diff
@@ -37,43 +47,45 @@ def command_create_profile(
     :param profile: The profile instance. This is an empty ``Profile`` instance created by the command line argument
         which currently only contains the selected profile name for the profile that is to be created.
     :param set_as_default: Whether to set the created profile as the new default.
+    :param email: Email for the default user.
+    :param first_name: First name for the default user.
+    :param last_name: Last name for the default user.
+    :param institution: Institution for the default user.
+    :param use_rabbitmq: Whether to configure RabbitMQ as the broker.
     :param kwargs: Arguments to initialise instance of the selected storage implementation.
     """
+    from aiida.brokers.rabbitmq.defaults import detect_rabbitmq_config
     from aiida.plugins.entry_point import get_entry_point_from_class
 
-    if not storage_cls.read_only and kwargs.get('email', None) is None:
+    if not storage_cls.read_only and email is None:
         raise click.BadParameter('The option is required for storages that are not read-only.', param_hint='--email')
 
-    email = kwargs.pop('email')
-    first_name = kwargs.pop('first_name')
-    last_name = kwargs.pop('last_name')
-    institution = kwargs.pop('institution')
-
     _, storage_entry_point = get_entry_point_from_class(storage_cls.__module__, storage_cls.__name__)
     assert storage_entry_point is not None
 
-    if kwargs.pop('use_rabbitmq'):
-        broker_backend = 'core.rabbitmq'
-        broker_config = {
-            key: kwargs.get(key)
-            for key in (
-                'broker_protocol',
-                'broker_username',
-                'broker_password',
-                'broker_host',
-                'broker_port',
-                'broker_virtual_host',
-            )
-        }
+    broker_backend = None
+    broker_config = None
+
+    if use_rabbitmq:
+        broker_config = detect_rabbitmq_config()
+
+        if broker_config is None:
+            echo.echo_warning(f'RabbitMQ server not reachable with default connection parameters: {broker_config}.')
```

Comment:
Note that I was quickly testing this in the field, and got the following

```python
Warning: RabbitMQ server not reachable with default connection parameters: None.
```

Since if the connection fails, the `detect_rabbitmq_config()` command returns `None`. Maybe it should actually return a `tuple` of the full config and if the connection was successful?

Reply:
Yeah, very sloppy on my part, apologies. I blame trying to fix these things while on holiday. I am not a fan of communicating success/error through return values, so I refactored to use exceptions instead.